p2pd/main.go

```diff
@@ -323,6 +339,21 @@ func main() {
 		opts = append(opts, libp2p.NoListenAddrs)
 	}
 
+	var securityOpts []libp2p.Option
+	if c.Security.SECIO {
+		securityOpts = append(securityOpts, libp2p.Security(secio.ID, secio.New))
+	}
+	if c.Security.TLS {
+		securityOpts = append(securityOpts, libp2p.Security(tls.ID, tls.New))
+	}
+	if c.Security.Noise {
+		securityOpts = append(securityOpts, libp2p.Security(noise.ID, noise.Maker()))
```

Comment:
This needs to be updated with the latest Noise.
```suggestion
		securityOpts = append(securityOpts, libp2p.Security(noise.ID, noise.New))
```

@yusefnapora IMO this branch is sufficient enough to get merged in. Noise isn't on by default so it's not an issue and would more easily enable ongoing interop testing. Can we get the dependencies updated in here so we can merge? I've been using this branch locally for interop testing and we have other contributors who this would be helpful for.